codebuild/bin/grep_simple_mistakes.sh

```diff
@@ -224,6 +224,56 @@ if [[ -n $S2N_ENSURE_WITH_INVALID_ERROR_CODE ]]; then
   printf "$S2N_ENSURE_WITH_INVALID_ERROR_CODE\n\n"
 fi
 
+#############################################
+# Assert tests don't specify the "default" security policy.
+#
+# Since the "default" policies are subject to change, tests should instead specify
+# an immutable numbered policy to avoid unwanted testing behavior.
+#############################################
+S2N_DEFAULT_SECURITY_POLICY_USAGE=$(find "$PWD" -type f -name "s2n*.c" -not -path "*/bindings/*" -not -path "*/bin/*")
+declare -A KNOWN_DEFAULT_USAGE
+KNOWN_DEFAULT_USAGE["$PWD/tls/s2n_config.c"]=1
+KNOWN_DEFAULT_USAGE["$PWD/tls/s2n_security_policies.c"]=1
+KNOWN_DEFAULT_USAGE["$PWD/tests/unit/s2n_security_policies_test.c"]=5
+KNOWN_DEFAULT_USAGE["$PWD/tests/unit/s2n_client_hello_test.c"]=2
+KNOWN_DEFAULT_USAGE["$PWD/tests/unit/s2n_connection_preferences_test.c"]=1
+KNOWN_DEFAULT_USAGE["$PWD/tests/unit/s2n_config_test.c"]=1
+# "default" string not used for specifying security policy
+KNOWN_DEFAULT_USAGE["$PWD/tests/unit/s2n_build_test.c"]=1
+KNOWN_DEFAULT_USAGE["$PWD/tests/unit/s2n_client_key_share_extension_test.c"]=2
```

Comment:
Could the search be made more specific to avoid false positives like these? I feel like it's not _that_ weird to write "default" in a context other than referencing the default security policy.

Like could we regex search for `set_cipher_preferences\(.*"default"\)` maybe?

Reply:
So we can get instances like [this test](https://github.com/aws/s2n-tls/blob/main/tests/unit/s2n_client_hello_test.c#L580), where the above regex would fail to match. Its hard to predict what other patterns developers might use in the future.

I therefore thing we should be a bit loose with the detection, especially since it only results in 2 false positives. It might worth adding a comment explaining this though if you agree.